service/controller/app/v1/resource_set.go

```diff
@@ -209,25 +210,44 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 		var k8sClient kubernetes.Interface
 		var g8sClient versioned.Interface
 
-		restConfig, err := kubeConfig.NewRESTConfigForApp(ctx, cr)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+		var isDeleting bool
```

Comment:
I see you took the shortcut. Things like this are the reason why I would like to get rid of the init ctx func. It is regularly abused like that. I made an effort to remove all logic from the init ctx func in the `aws-operator` already. It would be nice if we could not add more logic here. I think we should have a resource for this. I do not understand the problem that you had with the resource you already implemented. 

We could have a resource for the namespace deletion and a resource for the kube config magic. Then the kube config resource uses the information about the namespace deletion to decide what to do, just like all other resources do now. Then the init ctx func would be empty I think. In `aws-operator` we have the same approach with AWS clients that we gather in a separate resource which puts the clients into the controller context.

Reply:
This has nothing to do with shortcuts. So I'd appreciate it if we don't use that using that kind of language in reviews.

We are trying to find a solution to a problem that exists in production and has been reported by 3 separate people just this week.

If we want to deprecate this functionality why is there no deprecation notice in operatorkit? How are Ji and I supposed to know this? To me that is a communication problem.

https://github.com/giantswarm/operatorkit/blob/1a065237841c6c15cc0a3005d3a29a1c8d490b95/framework/framework.go#L19-L25

If this solves the issue in production I'd like us to merge it. We will then do a follow up PR to address these concerns and so we can deprecate this functionality from operatorkit.

cc @puja108

Reply:
It was not my intention to trigger your frustration here. I approved the PR twice already and do not block anything. So I assume I am not the root cause for whatever overwhelms you now. Do what you think is right and let me know when I can help with anything. I only provided suggestions as always, and not because I want to control the progress here, but because I have been asked for it.

Reply:
I put a card in the SIG Operator Board to document the init ctx func magic. You are right that this is likely not be well documented.

Reply:
Yes I got frustrated because I don't think we were taking a shortcut. So I wanted to push back a bit on that but I think my reaction was an overreaction and not helpful. So I'm sorry about that. ❤️ 

I like the design you're proposing and as always I really appreciate getting feedback that lets us improve what we do.

Reply:
After merging this, I will open another PR to migrate from initCtxFunc to another resource. Thanks for your input!

Reply:
Great thanks @tomahawk28 much appreciated and sorry I let the discussion get heated in your PR. ❤️